src/svg/qsvggenerator.cpp

```diff
@@ -998,7 +1017,36 @@ void QSvgPaintEngine::updateState(const QPaintEngineState &state)
     if (d->afterFirstUpdate)
         *d->stream << "</g>\n\n";
 
-    *d->stream << "<g ";
+	QPainter* p = painter();
+	if (p->hasClipping()) {
+		std::string clip_path = "";
+		QPainterPath path = p->clipPath();
+
+		if (path.elementCount() > 0) {
+			QPainterPath::Element starting_point = path.elementAt(0);
+			clip_path.append("<path d=\"M " + std::to_string(starting_point.x) + " " + std::to_string(starting_point.y) + " ");
+
+			for (int i = 1; i < path.elementCount(); i++) {
+				QPainterPath::Element element = path.elementAt(i);
+				std::string point = "L" + std::to_string(element.x) + ", " + std::to_string(element.y) + " ";
+				clip_path.append(point);
+			}
+
+			bool key_exists = clip_path_to_id.count(clip_path);
```

Comment:
I would remove key_exists and just check `if (0 == clip_path_to_id.count(clip_path))`